app/__init__.py

```diff
@@ -0,0 +1,33 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100, hidden: bool = False):
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+    @property
+    def is_alive(self) -> bool:
+        return self.health > 0
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
Code Efficiency issue: The `hide` method is correct, but it's more efficient to change the `bool` value in one line with `self.hidden = not self.hidden`.